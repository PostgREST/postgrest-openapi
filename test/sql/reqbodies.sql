-- Tables

-- defines an application/json request body
select jsonb_pretty(get_openapi_document('{test}')->'components'->'requestBodies'->'products'->'content'->'application/json');

-- defines an application/x-www-form-urlencoded request body
select jsonb_pretty(get_openapi_document('{test}')->'components'->'requestBodies'->'products'->'content'->'application/x-www-form-urlencoded');

-- defines a text/csv request body
select jsonb_pretty(get_openapi_document('{test}')->'components'->'requestBodies'->'products'->'content'->'text/csv');

-- Views

-- defines an application/json request body
select jsonb_pretty(get_openapi_document('{test}')->'components'->'requestBodies'->'big_products'->'content'->'application/json');

-- defines an application/x-www-form-urlencoded request body
select jsonb_pretty(get_openapi_document('{test}')->'components'->'requestBodies'->'big_products'->'content'->'application/x-www-form-urlencoded');

-- defines a text/csv request body
select jsonb_pretty(get_openapi_document('{test}')->'components'->'requestBodies'->'big_products'->'content'->'text/csv');

-- does not define a request body for non auto-updatable views
select get_openapi_document('{test}')->'components'->'requestBodies' ? 'non_auto_updatable' as value;
