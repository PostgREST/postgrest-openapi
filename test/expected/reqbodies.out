-- Tables
-- defines an application/json request body
select jsonb_pretty(get_openapi_document('{test}')->'components'->'requestBodies'->'products'->'content'->'application/json');
                        jsonb_pretty                         
-------------------------------------------------------------
 {                                                          +
     "schema": {                                            +
         "oneOf": [                                         +
             {                                              +
                 "$ref": "#/components/schemas/products"    +
             },                                             +
             {                                              +
                 "type": "array",                           +
                 "items": {                                 +
                     "$ref": "#/components/schemas/products"+
                 }                                          +
             }                                              +
         ]                                                  +
     }                                                      +
 }
(1 row)

-- defines an application/x-www-form-urlencoded request body
select jsonb_pretty(get_openapi_document('{test}')->'components'->'requestBodies'->'products'->'content'->'application/x-www-form-urlencoded');
                  jsonb_pretty                   
-------------------------------------------------
 {                                              +
     "schema": {                                +
         "$ref": "#/components/schemas/products"+
     }                                          +
 }
(1 row)

-- defines a text/csv request body
select jsonb_pretty(get_openapi_document('{test}')->'components'->'requestBodies'->'products'->'content'->'text/csv');
       jsonb_pretty        
---------------------------
 {                        +
     "schema": {          +
         "type": "string",+
         "format": "csv"  +
     }                    +
 }
(1 row)

-- Views
-- defines an application/json request body
select jsonb_pretty(get_openapi_document('{test}')->'components'->'requestBodies'->'big_products'->'content'->'application/json');
                          jsonb_pretty                           
-----------------------------------------------------------------
 {                                                              +
     "schema": {                                                +
         "oneOf": [                                             +
             {                                                  +
                 "$ref": "#/components/schemas/big_products"    +
             },                                                 +
             {                                                  +
                 "type": "array",                               +
                 "items": {                                     +
                     "$ref": "#/components/schemas/big_products"+
                 }                                              +
             }                                                  +
         ]                                                      +
     }                                                          +
 }
(1 row)

-- defines an application/x-www-form-urlencoded request body
select jsonb_pretty(get_openapi_document('{test}')->'components'->'requestBodies'->'big_products'->'content'->'application/x-www-form-urlencoded');
                    jsonb_pretty                     
-----------------------------------------------------
 {                                                  +
     "schema": {                                    +
         "$ref": "#/components/schemas/big_products"+
     }                                              +
 }
(1 row)

-- defines a text/csv request body
select jsonb_pretty(get_openapi_document('{test}')->'components'->'requestBodies'->'big_products'->'content'->'text/csv');
       jsonb_pretty        
---------------------------
 {                        +
     "schema": {          +
         "type": "string",+
         "format": "csv"  +
     }                    +
 }
(1 row)

-- does not define a request body for non auto-updatable views
select get_openapi_document('{test}')->'components'->'requestBodies' ? 'non_auto_updatable' as value;
 value 
-------
 f
(1 row)

